MayssaraOmar/assignment3/Word search/src/dictionary.h

```diff
@@ -0,0 +1,40 @@
+#include <unordered_map>
+#include <memory>
+#include <utility>
+#include <vector>
+#include <string>
+#include <cstdlib>
+using namespace std;
+
+
+#ifndef DICTIONARY_H_
+#define DICTIONARY_H_
+
+class dictionary {
+	struct node {
+		unordered_map<char, unique_ptr<node> > children;
+		bool isEndWord;
+		node() {
+			isEndWord = false;
+		}
+	};
+	unique_ptr<node> root;
+
+public:
+	dictionary();
+	//~dictionary(); no need for a destructor as we use smart pointers right?
```

Comment:
That is correct!
The other reason you might need a destruction is if this class has any virtual function (which in this case it does not).